build_tools/cmake/riscv.toolchain.cmake

```diff
@@ -28,17 +27,31 @@ set(CMAKE_CXX_COMPILER "${RISCV_TOOLCHAIN_ROOT}/bin/clang++")
 set(CMAKE_AR "${RISCV_TOOLCHAIN_ROOT}/bin/llvm-ar")
 set(CMAKE_RANLIB "${RISCV_TOOLCHAIN_ROOT}/bin/llvm-ranlib")
 set(CMAKE_STRIP "${RISCV_TOOLCHAIN_ROOT}/bin/llvm-strip")
-set(CMAKE_SYSTEM_LIBRARY_PATH "${RISCV_TOOLCHAIN_ROOT}/sysroot/usr/lib")
 
 set(RISCV_COMPILER_FLAGS)
 set(RISCV_COMPILER_FLAGS_CXX)
 set(RISCV_COMPILER_FLAGS_DEBUG)
 set(RISCV_COMPILER_FLAGS_RELEASE)
-set(RISCV_LINKER_FLAGS "-lstdc++ -lpthread -lm -ldl")
+set(RISCV_LINKER_FLAGS)
 set(RISCV_LINKER_FLAGS_EXE)
 
 if(RISCV_CPU STREQUAL "rv64")
+  set(CMAKE_SYSTEM_NAME Linux)
+  set(CMAKE_SYSTEM_LIBRARY_PATH "${RISCV_TOOLCHAIN_ROOT}/sysroot/usr/lib")
   list(APPEND RISCV_COMPILER_FLAGS "-march=rv64gc -mabi=lp64d")
+  set(RISCV_LINKER_FLAGS "${RISCV_LINKER_FLAGS} -lstdc++ -lpthread -lm -ldl")
+elseif(RISCV_CPU STREQUAL "rv32-baremetal")
+  set(CMAKE_SYSTEM_NAME Generic)
+  set(CMAKE_C_STANDARD 11)
+  set(CMAKE_C_EXTENSIONS Off)     # Force the usage of _ISOC11_SOURCE
+  set(IREE_BUILD_BINDINGS_TFLITE OFF CACHE BOOL "" FORCE)
+  set(IREE_BUILD_BINDINGS_TFLITE_JAVA OFF CACHE BOOL "" FORCE)
+  set(IREE_HAL_DRIVERS_TO_BUILD "" CACHE STRING "" FORCE)
+  set(IREE_TARGET_BACKENDS_TO_BUILD "DYLIB-LLVM-AOT;VMVX" CACHE STRING "" FORCE)
+  set(CMAKE_SYSTEM_LIBRARY_PATH "${RISCV_TOOLCHAIN_ROOT}/riscv32-unknown-elf/lib")
+  set(IREE_ENABLE_THREADING OFF CACHE BOOL "Builds IREE in with thread library support." FORCE)
```

Comment:
Is the description string needed here? Seems weird to have it say "build with support" when the whole line is setting that to `OFF`

Reply:
Description updated.